### PR TITLE
BREAKING! Replace `transfer_api` with latest TFLite usage

### PR DESCRIPTION
**Pull request by SichangHe**
This PR rewrites the "`.tflite` file generation section" and the "on-device training section" following the latest [On-Device Training with TensorFlow Lite](https://www.tensorflow.org/lite/examples/on_device_training/overview).

## Main features

- Support regression.
- Support custom loss and accuracy on the client side.
- Simplify and consolidate `.tflite` file generation process.

## Breakage

- Rewrote backend database migrations, so the database needs to be reinitialized.
- Changed `org.eu.fedcampus.train.db.TFLiteModel.Model`, so the client database either needs a `AutoMigration` or deletion by reinstalling the app.
    - `flwr.android_client.Db` is reset to `version = 1`. So, the demo could only run after the App is deleted and reinstalled.

### `.tflite` file generation `gen_tflite`

- Generate only 1 `.tflite` file instead of 5.
- `BaseTFLiteModel` to inherit from and `tflite_model_class` to annotate with.
    - Allows custom input and output shape without flatten.

### Backend database `train.models`

- `TFLiteModel` tracks the byte size of the layers.
- `ModelParams` stores `params` as Pickle instead of JSON.

### Client package `org.eu.fedcampus.train`

- `Train` is now generic on the input and output of the model.
    - Does not care about classifications anymore. Classification is left for the user to implement (it's just $\arg\max$).
    - Provide a `modelFile` to user instead of `modelLoader`.
- `FlowerClient` directly calls the TFLite `Interpreter` instead of using `TransferLearningModelWrapper`.
    - Manages `trainingSamples` and `testSamples` with `ReentrantReadWriteLock`.
    - `FlowerClient.addSample` is the synchronous, thread-safe, blocking method to load training and test data.

### Example demo client

- The database need to be reset by uninstalling the app because the schemas are reset.
- `loadData` takes `FlowerClient` instead of the no-longer-used `TransferLearningModelWrapper`.
    - Other users should load data directly into `FlowerClient` as well.
    - `prepareImage` 

## Note on the removed implementation

The original on-device training code in Flower Android example was copied from a TensorFlow 2019 example. Then, we copied it. This PR removed it.

## TODOs:

- [x] Documentation.

### CODE REVIEW

**Comment by SichangHe** [comment]
New CIFAR10 TFLite file using new infrastructure: 
[cifar10.zip](https://github.com/FedCampus/dyn_flower_android_drf/files/11858642/cifar10.zip)
